src/plugins/data/common/field_formats/utils/deserialize.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { identity } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import {
+  convertDateRangeToString,
+  DateRangeKey,
+} from '../../../../../legacy/ui/public/agg_types/buckets/date_range';
+import {
+  convertIPRangeToString,
+  IpRangeKey,
+} from '../../../../../legacy/ui/public/agg_types/buckets/ip_range';
```

Comment:
These will also break when #56353 merges; the BWC for imports from `agg_types` only supports imports from the top level `ui/agg_types`:
```ts
import {
  convertDateRangeToString,
  DateRangeKey,
  convertIPRangeToString,
  IpRangeKey,
} from 'ui/agg_types';
```